src/agent/agent.c

```diff
@@ -602,6 +677,32 @@ bool agent_start(Agent *agent) {
                 return false;
         }
 
+        r = sd_bus_call_method(
+                        agent->systemd_dbus,
+                        SYSTEMD_BUS_NAME,
+                        SYSTEMD_OBJECT_PATH,
+                        SYSTEMD_MANAGER_IFACE,
+                        "Subscribe",
+                        &error,
+                        &m,
+                        "");
+        if (r < 0) {
+                fprintf(stderr, "Failed to issue subscribe call: %s\n", error.message);
+                sd_bus_error_free(&error);
+                return false;
+        }
+
+        r = sd_bus_add_match(
```

Comment:
Could we use `sd_bus_match_signal` instead? I think that would be a bit easier to read.

Reply:
No, sd_bus_match_signal() is a limited version of add_match() which doesn't support the `path_namespace` matching. And we need this here, because we want to match all object paths like `/org/freedesktop/systemd/job/*`. If we were to use match_signal with an empty object_path we would get matches for all property changes in any path.

Reply:
Oh, I confused the `path` parameter in `sd_bus_match_signal` with the `path_namespace`.

Reply:
Yeah, AddMatch (https://dbus.freedesktop.org/doc/dbus-specification.html#bus-messages-add-match) supports both path and path_namespace match rules (https://dbus.freedesktop.org/doc/dbus-specification.html#message-bus-routing-match-rules)

Reply:
Looking at that, we could perhaps even add arg0=`org.freedesktop.systemd1.Job` to limit callbacks to property notifications for the properties of that particular interface. Not sure it really matters here though, as no other interfaces with properties are implemented by the job object.